prowler/providers/gcp/gcp_provider_new.py

```diff
@@ -0,0 +1,119 @@
+import os
+import sys
+from typing import Any, Optional
+
+from colorama import Fore, Style
+from google import auth
+from google.oauth2.credentials import Credentials
+from googleapiclient import discovery
+
+from prowler.lib.logger import logger
+from prowler.providers.common.provider import CloudProvider
+
+
+class GcpProvider(CloudProvider):
```

Comment:
We should do something like `GCPProvider`, `AzureProvider`, `AWSProvider`, what do you think?